From 72b02e5b464289d6ab11ac7b0cc023418ed03100 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Thu, 18 Feb 2021 20:10:01 +0500
Subject: [PATCH 1/3] Enable debug info in release profile

---
 Cargo.toml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index 783ae09716e4..a6584fcf39e9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -86,3 +86,7 @@ resolver = "2"
 
 [profile.dev]
 split-debuginfo = "unpacked"
+
+[profile.release]
+opt-level = 3
+debug = true
-- 
2.34.1

