From cddaab19d0cac3aad563a4104990a324a329cc28 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Thu, 18 Feb 2021 20:10:01 +0500
Subject: [PATCH 1/3] Enable debug info in release profile

---
 Cargo.toml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index 9aad90aeb4f1..c431845eefe5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -94,3 +94,7 @@ exclude = [
 
 # This prevents a Travis CI error when building for Windows.
 resolver = "2"
+
+[profile.release]
+opt-level = 3
+debug = true
-- 
2.37.2

