From ddc764d4b57ea0b009710abb045915902c934c0e Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Thu, 7 Oct 2021 07:57:32 +0500
Subject: [PATCH 2/3] Enable LTO

---
 Cargo.toml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index bb8fe69a3a34..19a68136e68b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,3 +92,5 @@ exclude = [
 [profile.release]
 opt-level = 3
 debug = true
+lto = "fat"
+codegen-units = 1
-- 
2.35.1

