From b1d5dbc5220a039b1a383d858843c43dfb84132d Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Thu, 7 Oct 2021 07:57:32 +0500
Subject: [PATCH 2/3] Enable LTO

---
 Cargo.toml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index 14f98617aa9b..9ac020237f4e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -95,3 +95,5 @@ resolver = "2"
 [profile.release]
 opt-level = 3
 debug = true
+lto = "fat"
+codegen-units = 1
-- 
2.35.1

