From ab89c0556ce665d1259d70ecda02fbedc22d739e Mon Sep 17 00:00:00 2001
From: buffalu <85544055+buffalu@users.noreply.github.com>
Date: Sun, 6 Oct 2024 08:26:26 -0500
Subject: [PATCH] Loosen tip requirement [v1.18] (#684)

---
 core/src/tpu.rs | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/core/src/tpu.rs b/core/src/tpu.rs
index 5f8942e086..624e64c233 100644
--- a/core/src/tpu.rs
+++ b/core/src/tpu.rs
@@ -290,11 +290,9 @@ impl Tpu {
 
         let bundle_account_locker = BundleAccountLocker::default();
 
-        // tip accounts can't be used in BankingStage to avoid someone from stealing tips mid-slot.
-        // it also helps reduce surface area for potential account contention
+        // The tip program can't be used in BankingStage to avoid someone from stealing tips mid-slot.
         let mut blacklisted_accounts = HashSet::new();
-        blacklisted_accounts.insert(tip_manager.tip_payment_config_pubkey());
-        blacklisted_accounts.extend(tip_manager.get_tip_accounts());
+        blacklisted_accounts.insert(tip_manager.tip_payment_program_id());
         let banking_stage = BankingStage::new(
             block_production_method,
             cluster_info,
